### PR TITLE
feat: bookings query endpoint

### PR DESCRIPTION
**Pull request by KevDev0247**
#### **What does this PR do?**

Created bookings query endpoint
- added bookings query controller
- add query route

#### **How is this PR tested?**

Tested with postman (no auth requirement for queries)

#### **Checklist**

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code in hard-to-understand areas
-   [x] My changes generate no new warnings
-   [x] My merge commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: src/routes/api/bookings/queryBookings.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Request, Response } from 'express';
+import Booking from '../../../models/booking.model';
+
+/**
+ * Route to query all bookings or the bookings from one booker
+ * @author Kevin Wang
+ */
+const queryBookingsRoute = async (req: Request, res: Response): Promise<void> => {
+    const bookerEmail = req.query.id;
```
Comment (sunny-zuo) [inline src/routes/api/bookings/queryBookings.ts:9]: This lets people arbitrarily view people's bookings - can we authenticate this request?

File: src/routes/api/bookings/queryBookings.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    if (bookerEmail) {
+        if (typeof bookerEmail === 'string') {
+            // If an booker's discord id is included in the query, retrieve the bookings of the and return the booking
+            const bookings = await Booking.find({ booker: bookerEmail }).catch((error) => {
+                res.status(404).json({ error });
```
Comment (sunny-zuo) [inline src/routes/api/bookings/queryBookings.ts:14]: `404` on an arbitrary error doesn't make sense, and the return statement on the next line will only exit the promise.

File: src/routes/api/bookings/queryBookings.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                return;
+            });
+            res.status(200).json({ bookings });
+        } else {
+            // If an booker's discord id is not valid, return error
+            res.status(404).json({ error: 'Validation failed, booker is not valid' });
+        }
+    } else {
+        // If no booker's discord id is included in the query, retrieve all the bookings and return the booking
```
Comment (sunny-zuo) [inline src/routes/api/bookings/queryBookings.ts:23]: Why do we need this endpoint?

**Review by sunny-zuo — CHANGES_REQUESTED** [review]

**Comment by Notoh** [comment]
Superseded by #32

**Pull request closed without merging**